dipy/reconst/peaks.py

```diff
@@ -157,8 +157,59 @@ def peak_directions(odf, sphere, relative_peak_threshold=.5,
 
 
 class PeaksAndMetrics(PeaksAndMetricsDirectionGetter):
-    pass
-
+    def rebuild_class_state(self, sphere, peak_dirs, peak_values, peak_indices,
+                    gfa_array, qa_array, shm_coeff, B, odf_array):
+        """
+        Helper function which helps unserialize a PeaksAndMetrics objects
+        """
+        self.sphere = sphere
+        self.peak_dirs = peak_dirs
+        self.peak_values = peak_values
+        self.peak_indices = peak_indices
+        self.gfa = gfa_array
+        self.qa = qa_array
+        self.shm_coeff = shm_coeff
+        self.B = B
+        self.odf = odf_array
+
+    def __setstate__(self, state):
+        """
+        For unpickling of the object
+        The state should be defined as follows: (qa, ind, odf_vertices)
+        """
+        #print("In __setstate__ method for PeaksAndMetrics dipy/reconst/peaks.py. Got to this method!")
+        
+        self.rebuild_class_state(*state["class_state"])
+        self.initialized = state["is_initialized"]
+
+
+        if self.initialized:
+            self._initialize()
+            current_state = state["current_state"]
+            self._qa = current_state["_qa"]
+            self._ind = current_state["_ind"]
+            self._odf_vertices = current_state["_odf_vertices"]
```

Comment:
These assignments are already done in `self._initialize()` so do we need to do them again?